packages/data-point/lib/reducer-entity/factory.js

```diff
@@ -19,7 +19,10 @@ function ReducerEntity () {
 module.exports.ReducerEntity = ReducerEntity
 
 function isEntity (source) {
-  return source.match(/^[^$]([\w.]+):?([\w.]+)(\[])?/) !== null
+  return (
+    typeof source === 'string' &&
+    source.match(/^[^$]([\w.]+):?([\w.]+)(\[])?/) !== null
```

Comment:
What are the rules for naming entities? Do they not always use the _<EntityType>:<EntityId>_ pattern? It surprises me to see that the colon is optional, so any string with three or more letters is a valid entity name. Also surprised to see that "aaa:" is valid and "a:aaa" is not valid (do entity types need to have more than one letter?).

Reply:
regex is not my strength, would be great if you send a PR to fix it? this change you see here is related to a prettier automatic change related to a change I made lines above

Reply:
to clarify, the only real change on this PR related to this chunk is line 23

Reply:
Ok! I will get cracking on a new regex

Reply:
#37